problem/views.py

```diff
@@ -232,14 +232,18 @@ def testcase(request, pid, tid=None):
         if 't_in' in request.FILES:
             TESTCASE_PATH = config_info.get_config('path', 'testcase_path')
             try:
-                with open('%s%s.in' % (TESTCASE_PATH, testcase.pk), 'w') as t_in:
+                input_filename = '%s%s.in' % (TESTCASE_PATH, testcase.pk)
+                output_filename = '%s%s.out' % (TESTCASE_PATH, testcase.pk)
+                with open(input_filename, 'w') as t_in:
                     for chunk in request.FILES['t_in'].chunks():
-                        t_in.write(chunk.replace('\r\n', '\n'))
-                    logger.info("testcase %s.in saved by %s" % (testcase.pk, request.user))
-                with open('%s%s.out' % (TESTCASE_PATH, testcase.pk), 'w') as t_out:
+                        t_in.write(chunk)
+                call(['dos2unix', input_filename])
```

Comment:
prefer to use ```check_call``` instend of call.
but you can customize the response of result with ```Popen``` function